### PR TITLE
Conversion tool documentation updates

### PR DESCRIPTION
**Pull request by drewmoseley**

### CODE REVIEW

**Comment by estenberg** [comment]
Nice improvement! Not sure if you are finished here, but leaving a quick review of current state.

A couple of comments:
* similar to mender-artifact, would refer to it as mender-convert (not Mender conversion tool)
* I see some section use OS as the top level (Device integration -> System requirement -> $OS -> Bootloader support), but most sections don't. Should we make OS the top level everywhere to make this consistent? I think it is rare that someone using Debian would be interested in how things work on Yocto and vice versa, thus a more top level OS would make sense IMO.
* Device Integration -> System requirement -> Yocto. Should be Yocto Project to stay consistent.
* Device Integration -> System requirement -> Yocto -> Yocto/Debian bootloader support. Don't need the OS in the section name of this leaf section, it is in the parent section.
* artifacts/building-mender-image/building-debian-image is too skinny. I will leave a quick howto I wrote for consideation in ticket, maybe some parts we can use of that

**Comment by drewmoseley** [comment]
@estenberg no I was not yet complete.  The reworking of the Menu entries for the Devices section was just added as https://github.com/mendersoftware/mender-docs/pull/520/commits/3c5927f44713c0fabbb932b1b0f82c06a640e0ad

Please review and let me know if that is what you were thinking; I don't want to spend more time on it right now in case I am off in the weeds.

**Comment by estenberg** [comment]
Yep, looks great!

Some minor comments that can be worked on over time:
* The P in Yocto Project should be capitalized
* Section names like "Yocto proejct requirements" are too long. When there is something OS-specific would simply state the OS, e.g. "Yocto Project". I think it is implicit from that what we are talking about (e.g. requirements, how to build artifacts, etc.) as it is captured in subsections.

**Comment by drewmoseley** [comment]
@estenberg I think this pretty much completes https://tracker.mender.io/browse/MEN-2148.  Please take a look.

File: 03.Devices/03.Debian-family/docs.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Debian family
+taxonomy:
+    category: docs
+---
+
+This section describes the requirements for Mender when integrated
+with the Debian family target OS images such as
+[Debian](https://www.debian.org/), [Ubuntu](https://www.ubuntu.com/)
+and [Raspbian](https://www.raspberrypi.org/downloads/raspbian/).
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/docs.md]: External links: Should append ?target=_blank to open in new tab.

File: 03.Devices/03.Debian-family/02.Bootloader-support/docs.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+title: Bootloader support
+taxonomy:
+    category: docs
+---
+
+Currently the [Mender conversion tool](https://github.com/mendersoftware/mender-conversion-tools) uses a modified U-Boot for Raspbian and an unmodified GRUB for Debian on Beaglebone. Choosing a different bootloader is not supported.
+
+These scripts require bootloader installation after repartitioning of original system image. To install the correct bootloader, execute the following:
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/02.Bootloader-support/docs.md]: Would remove from here "These scripts require..." until end of doc.

This section is about requirements and what is supported, not a tutorial on how to install the bootloader.
Also, I don't see a workflow where people would clone this tool just to install the bootloader, and that's it. Thus the below should rather be in a more full-blown tutorial (somewhere else).

File: 03.Devices/03.Debian-family/02.Bootloader-support/docs.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+title: Bootloader support
+taxonomy:
+    category: docs
+---
+
+Currently the [Mender conversion tool](https://github.com/mendersoftware/mender-conversion-tools) uses a modified U-Boot for Raspbian and an unmodified GRUB for Debian on Beaglebone. Choosing a different bootloader is not supported.
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/02.Bootloader-support/docs.md]: External link, open new tab.

File: 03.Devices/03.Debian-family/02.Bootloader-support/01.GRUB/docs.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 ---
-title: GRUB with Debian
+title: GRUB
 taxonomy:
     category: docs
 ---
 
-[Mender conversion scripts](https://github.com/mendersoftware/mender-conversion-tools) use GRUB as second stage bootloader for Debian on Beaglebone. Mender conversion tools build GRUB from [official repository](https://www.gnu.org/software/grub/grub-download.html). Mender project does not require any patches for GRUB. GRUB should be built with EFI platform support. This command will install GRUB bootloader for disk image with partition layout compliant to Mender requirements:
+The [Mender conversion tool](https://github.com/mendersoftware/mender-conversion-tools) uses GRUB as second stage bootloader for Debian on Beaglebone. This tool build GRUB from the [official repository](https://www.gnu.org/software/grub/grub-download.html). Mender does not require any patches for GRUB and should be built with EFI platform support. Execute the following to install the GRUB bootloader into a disk image with partition layout compliant with Mender requirements:
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/02.Bootloader-support/01.GRUB/docs.md]: Exsternal link

Comment (estenberg) [reply 03.Devices/03.Debian-family/02.Bootloader-support/01.GRUB/docs.md]: Simiarly, would drop everything from "Execute the following to install.." to the end of doc. We should talk about how it conceptually works and requirements only here, not how to convert something.

File: 03.Devices/03.Debian-family/02.Bootloader-support/02.U-Boot/docs.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Debian integration with U-Boot
+title: U-Boot
 taxonomy:
     category: docs
 ---
 
-[Mender conversion scripts](https://github.com/mendersoftware/mender-conversion-tools) provide script for building and installing patched U-Boot for Raspbian. Currently only patched U-Boot is supported by Mender on RaspberryPi 3. This command will install bootloader on disk image with partition layout compliant with Mender requirements.
+The [Mender conversion tool](https://github.com/mendersoftware/mender-conversion-tools) provides for building and installing patched U-Boot for Raspbian. Currently only patched U-Boot is supported by Mender on Raspberry Pi 3. This command will install bootloader on disk image with partition layout compliant with Mender requirements.
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/02.Bootloader-support/02.U-Boot/docs.md]: Similar. Drop from "This command will...".

Maybe link to the U-Boot version we are using? I think this is from a repo in github/mendersoftware?

File: 04.Artifacts/01.Yocto-project/docs.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+title: Yocto artifacts
```
Comment (estenberg) [inline 04.Artifacts/01.Yocto-project/docs.md]: Yocto Project

File: 04.Artifacts/01.Yocto-project/docs.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+taxonomy:
+    category: docs
+---
+This section outlines the steps for working with Mender artifacts created by the [Yocto project](http://www.yoctoproject.org/) builds.
```
Comment (estenberg) [inline 04.Artifacts/01.Yocto-project/docs.md]: External link

**Comment by drewmoseley** [comment]
@estenberg updated per your review comments. I think I addressed all points.  Please review.

**Comment by estenberg** [comment]
Only issue I can see is that Aritfacts -> Variables is still not under Yocto? Was this intentional?

**Comment by drewmoseley** [comment]
Fixed.

**Pull request merged**